app/api/routers/breeze_buddy/templates/handlers.py

```
@@ -424,3 +429,96 @@ async def replace_template_handler(
+
+        if blockers:
+            detail = (
+                f"Template '{existing_template.name}' cannot be safely deleted. "
+                f"Reasons: {'; '.join(blockers)}. "
+                f"Please remove these references before deleting the template."
+            )
+            logger.warning(f"Template {template_id} deletion blocked: {detail}")
+            raise HTTPException(
+                status_code=status.HTTP_409_CONFLICT,
+                detail=detail,
+            )
```

Ruff (0.15.2)

[warning] Abstract `raise` to an inner function

(TRY301)